src/main/java/com/google/sps/servlets/ProfileServlet.java

```diff
@@ -78,7 +78,7 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
         if (dairyFree) {
           profileObjectBuilder.setDairyFree(dairyFree);
         }
-        if (allergies != null && allergies.length > 0) {
+        if (allergies != null && allergies.size() > 0) {
```

Comment:
Do you need both checks? I'm not sure just wondering

Reply:
.size() gives an error when allergies == null.